File: docs/examples/unit_tests/student_solution/java/Buyer_mistake.java
import java.io.*;
class Buyer
{
  intentional mistake;
  private int cash;

  public Buyer(int cash)
  {
    this.cash=cash;
  }

  public int getCash()
  {
    return cash;
  }

  public void setCash(int cash)
  {
    this.cash=cash;
  }

  public void buy(Seller x)
  {
    if(cash>0)
    {
      if(x.sell()==1)
      {
        cash--;
      }
    }
  }
}

File: tests/test_modules/data/scoreboard/BuyerMistake.cpp
#include "Buyer.hpp"

intentional error;
Buyer::Buyer(int cash)
{
    this->cash=cash;
}

int Buyer::getCash()
{
    return cash;
}

void Buyer::setCash(int cash)
{
    this->cash=cash;
}

void Buyer::buy(Seller x)
{
    if(cash>0)
    {
      if(x.sell()==1)
      {
        cash--;
      }
    }
}
